src/core/api/coap_api.cpp

```diff
@@ -66,6 +66,11 @@ otError otCoapHeaderAppendUintOption(otCoapHeader *aHeader, uint16_t aNumber, ui
     return static_cast<Coap::Header *>(aHeader)->AppendUintOption(aNumber, aValue);
 }
 
+otError otCoapHeaderSetContentFormatOption(otCoapHeader *aHeader, otCoapOptionContentFormat aContentFormat)
+{
+    return otCoapHeaderAppendUintOption(aHeader, OT_COAP_OPTION_CONTENT_FORMAT, (uint32_t)aContentFormat);
```

Comment:
Also, there already is Header::AppendContentFormatOption function, which is not used anywhere in the code and has empty enum MediaType. Looking at current code style, suggests that MediaType should be replaced with your new otCoapOptionContentFormat enum and a call to otCoapHeaderAppendUintOption replaced to Header::AppendContentFormatOption. However @jwhui  should have the last word here.